stellargraph/core/graph_networkx.py

```diff
@@ -39,6 +39,9 @@
 from .utils import is_real_iterable
 
 
+NeighbourWithWeight = namedtuple("NeighbourWithWeight", ["node", "weight"])
```

Comment:
@kjun9 I think in this `namedtuple("NeighbourWithWeight", ["node", "weight"])` the label "weight" is hard coded? When I create a graph with a different attribute name and try to access edge weights, I get None. 
Here is a small example: 

```
G = nx.MultiDiGraph()
G.add_edge('a', 'b',  t = 1, label='temporal')
G.add_edge('a', 'c', t=2, label='temporal')
G.add_edge('c', 'd', t=3, label='temporal')
SG = sg.StellarGraph(G)
print(SG.info())
list(SG.neighbors('a',include_edge_weight=True))
list(SG.neighbors('c', include_edge_weight = True))
list(SG.in_nodes('c', include_edge_weight = True))
list(SG.out_nodes('c', include_edge_weight = True))

```
Also, 
`edge_weight` doesn't give back the weights. 
`[ (n, u, SG.edge_weights(n, u)) for u in SG.neighbors('a')]`

If the attribute is named `weight` all the methods work.

Reply:
I think there's actually two different things here:

* If you want to assign a different networkx attribute as the "edge weight", the following should work:
```
>>> import networkx as nx
>>> nxg = nx.MultiDiGraph()
>>> nxg.add_edge('a', 'b', t=1, label="temporal")
0
>>> nxg.add_edge('a', 'c', t=2, label='temporal')
0
>>> nxg.add_edge('c', 'd', t=3, label='temporal')
0
>>> import stellargraph as sg
>>> g = sg.StellarGraph(nxg, edge_weight_label="t")
>>> print(g.info())
NetworkXStellarGraph: Directed multigraph
 Nodes: 4, Edges: 3

 Node types:
  default: [4]
    Edge types: default-temporal->default

 Edge types:
    default-temporal->default: [3]
        Attributes: {'t'}

>>> list(g.neighbors('a', include_edge_weight=True))
[NeighbourWithWeight(node='b', weight=1), NeighbourWithWeight(node='c', weight=2)]
```
but yes, as @huonw points out, the `edge_weight_label` parameter was undocumented so it would've been difficult to figure out the above
* I'd say that the `"weight"` within the named tuple is hardcoded, the same way the term "weight" is hardcoded in the parameter `include_edge_weight` - currently `StellarGraph` only supports weights on edges **by design** and **not** an arbitrary set of attributes on edges, but the user has the ability to specify `edge_weight_label` parameter to decide which networkx edge attribute gets used as the "weight". 

I think for now, for the temporal walk use-case, we'd just have to remember that we'd be using the term `weight` to refer to timestamps. If you think this is going to be too jarring, we'd have to consider supporting arbitrary edge attributes. What do you think?

Reply:
We can stick to the term "weight", that would work as long as we know it. Also, for walks we don't need multiple attributes over the edges. So  that isn't necessary either.